tests/integ/ml/test_transformer.py

```diff
@@ -0,0 +1,505 @@
+            ),
+        ],
+    )
+    # merge and sparse is false
+    res = model.transform(df, sparse=False)
+    Utils.check_answer(
+        res,
+        [
+            Row(A=1, B=4, C=5, ONEHOTARRAY="[\n  0,\n  1,\n  0,\n  1,\n  0\n]"),
+            Row(A=2, B=3, C=6, ONEHOTARRAY="[\n  1,\n  0,\n  1,\n  0,\n  1\n]"),
+            Row(A=2, B=4, C=5, ONEHOTARRAY="[\n  1,\n  1,\n  0,\n  1,\n  0\n]"),
+        ],
+    )
+
+    # merge and sparse is true
+    onehotencoder.output_cols = ["res_a", "res_b", "res_c"]
+    res = model.transform(df)
+    Utils.check_answer(
+        res,
+        [
+            Row(
+                A=1,
+                B=4,
+                C=5,
+                RES_A="[\n  1,\n  1\n]",
+                RES_B="[\n  2,\n  0,\n  1\n]",
+                RES_C="[\n  2,\n  0,\n  1\n]",
+            ),
+            Row(
+                A=2,
+                B=3,
+                C=6,
+                RES_A="[\n  1,\n  0,\n  1\n]",
+                RES_B="[\n  2,\n  1,\n  1\n]",
+                RES_C="[\n  2,\n  1,\n  1\n]",
+            ),
+            Row(
+                A=2,
+                B=4,
+                C=5,
+                RES_A="[\n  1,\n  0,\n  1\n]",
+                RES_B="[\n  2,\n  0,\n  1\n]",
+                RES_C="[\n  2,\n  0,\n  1\n]",
+            ),
+        ],
+    )
+    # merge and sparse is false
+    res = model.transform(df, sparse=False)
+    Utils.check_answer(
+        res,
+        [
+            Row(
+                A=1,
+                B=4,
+                C=5,
+                RES_A="[\n  0\n]",
+                RES_B="[\n  1,\n  0\n]",
+                RES_C="[\n  1,\n  0\n]",
+            ),
+            Row(
+                A=2,
+                B=3,
+                C=6,
+                RES_A="[\n  1\n]",
+                RES_B="[\n  0,\n  1\n]",
+                RES_C="[\n  0,\n  1\n]",
+            ),
+            Row(
+                A=2,
+                B=4,
+                C=5,
+                RES_A="[\n  1\n]",
+                RES_B="[\n  1,\n  0\n]",
+                RES_C="[\n  1,\n  0\n]",
+            ),
+        ],
+    )
+
+
+def test_unknown_category_given_by_transformer_onehotenoder(session):
+    df = session.createDataFrame(
+        [[1, 4, 5], [2, 4, 5], [2, 3, 6]], schema=["a", "b", "c"]
+    )
+    onehotencoder = OneHotEncoder()
+    onehotencoder.input_cols = ["A", "B", "C"]
+    onehotencoder.output_cols = ["ONEHOTARRAY"]
+    model = onehotencoder.fit(df)
+    # test category table
+    df_table = session.table(model._category_table_name["A"])
+    Utils.check_answer(
+        df_table,
+        [Row(STATES_A=1, STATES_A_ENCODER=0), Row(STATES_A=2, STATES_A_ENCODER=1)],
+    )
+    df_unknown_value = session.createDataFrame(
+        [[6, 4, 5], [2, 4, 5], [2, 3, 6]], schema=["a", "b", "c"]
+    )
+    res = model.transform(df_unknown_value)
+    Utils.check_answer(
+        res,
+        [
+            Row(
```

Comment:
It looks like the row order is not preserved?

Reply:
row order is preserved, but util.check_answer accept random order of answer